### PR TITLE
Get Folder - #1

### LINKED ISSUE #1: Prompt for folder name

### PR DESCRIPTION
**Pull request by KaraBeason**
Closes #1 .  I re-did my unit tests, modeling them after Michelle's, and restructured my function in Scatt.java to take a file object so the file obj is created in main instead of the helper function.

@haietza I noticed while testing mine that the testZip and testUnzip unit tests are failing intermittently, any idea why? The issue is present on master for me too.

Also, I think this was more accurately a medium than a hard issue, so I'm going to change the label unless anyone objects.

### CODE REVIEW

File: Scatt.java [hunk 1]
```diff
@@ -24,5 +26,28 @@
...
+        Boolean isValid = readValidDirectory(directory);
+    }
+
+    /**
+     * Check to see if folder is valid.
+     *
+     * @param dir directory file object
+     * @return true or false
+     */
+    public static Boolean readValidDirectory(File dir)
```
Comment (haietza) [inline Scatt.java]: Should we make this private, since only we need to call it from main?

File: Scatt.java [hunk 2]
```diff
@@ -24,5 +26,28 @@
...
+    {
+        if (dir.exists() || dir.isDirectory())
```
Comment (haietza) [inline Scatt.java]: Should this be && - don't we want to ensure that the directory exists AND that it is a directory?

File: ScattTest.java [hunk 3]
```diff
@@ -16,4 +24,54 @@
  */
 public class ScattTest
 {
+    public Scatt testScatt;
```
Comment (haietza) [inline ScattTest.java]: Does this field need to be public?

File: ScattTest.java [hunk 4]
```diff
@@ -16,4 +24,54 @@
...
+        assertFalse(testScatt.readValidDirectory(testDir));
+    }
+
+    /**
+     * Tear down after tests.
+     */
+    @After
+    public void tearDown()
+    {
+        testDir.delete();
```
Comment (haietza) [inline ScattTest.java]: Should we add testScatt = null to the tearDown process also?

File: ScattTest.java [hunk 5]
```diff
@@ -1,3 +1,11 @@
+import org.junit.Test;
+import java.io.File;
+import java.io.IOException;
```
Comment (haietza) [inline ScattTest.java]: Checkstyle reports ScattTest.java:3:8: Unused import - java.io.IOException

File: Scatt.java (hunk 1 above)
Comment (KaraBeason) [inline Scatt.java]: If it's private I can't call it from the unit tests.

Comment (haietza) [inline Scatt.java]: Ah - gotcha!

**Comment by KaraBeason** [comment]
@haietza I think I got all of the changes!  LMK your thoughts :)

**Comment by haietza** [comment]
@KaraBeason Looks great! I'll check into the unit test failure issue and get that pushed up ASAP. Thanks!

**Pull request merged**